Chrysalis-Papilio/Soen490-Papilio-Website
# Papilio 45 delete employee

## General info
Added a Delete Employee button in the employee dashboard which allow the administrator to delete the selected employee/s.

The delete feature is in a checklist form which display the existing employees and each row can be checked . User can select one employee or more and then click on the delete button to submit the request. Once the request is submitted , the selected employee/s will be deleted from the list.

In order to have the dynamic checklist functioning , a modification in the employee table was required to change employee data from static to dynamic.

_Issue number_: Papilio-45

## Manual Test

To test the functionality :

Since the the delete button is visible only to employee who has the admin role , and you might not have this role .A temporary workaround to test the the button is to comment out the statement :
[`if (employee.role !== 'Admin') { return <></>; }` in EmployeeDashboard](https://github.com/Chrysalis-Papilio/Soen490-Papilio-Website/blob/ac98e99e6f95c177a6729dad5aa1d6380ec194c8/src/pages/Dashboard/Employees/index.tsx#L74-L76) so the buttons can be seen for all employees.
Make sure that you have the appropriate variables in the backend .env file.

1. Start the backend
`npm run build && npm start`
2. Start the website
`npm start`
3. Navigate to [login page](http://localhost:3000/login)
4. Login with a user (contact @p-carrier to get one)
5. Navigate to Employee Page of the dashboard
6. Click on Delete employee
7. select the employees that needs to be deleted
8. click on the delete employee button to submit
the expected results is the selected employee/s are deleted from the list.

## Test
run `npm test`
